### PR TITLE
Release 1.0

### PR DESCRIPTION
**Pull request by mcupak**
Last chance for 1.0 requests before we tag from `master`. Please speak now or forever hold your peace.

### CODE REVIEW

File: USECASES.md [hunk 1]
```diff
@@ -1,117 +1,117 @@
...
+
+## Data types
+The Data Connect API is not expected to define data types, but it will be important
+that some shared vocabulary are available to facilitate federated queries.
+
+All responses in the feature matrix indicated that the availability of a
+standard set of fields is "nice to have" or "must have", while the ability to
+have arbitrary non-standard fields is a "must have".
+
+Data type mentioned in the responses include:
```
Comment (jfuerth) [inline USECASES.md]: Minor: "Data type" should be plural

File: spec/api.yaml [hunk 2]
```diff
@@ -0,0 +1,269 @@
...
+    get:
+      summary: Get a Table
+      description: Returns the information about the Table identified by name.
+      operationId: getTable
+      tags:
+        - Table API
+      parameters:
+        - name: table_name
+          in: path
+          description: Unique Table name, unique to this implementation
```
Comment (jfuerth) [inline spec/api.yaml]: Minor: this wording is a bit awkward, and it's repeated several times in this doc.

Maybe better to say "A table name"?

We could mention the unique thing on the `Table.name` property. We could also provide examples there such as `my_schema.my_table` and `my_catalog.my_schema.my_table` to make it clear that these are unique, qualified names.

File: spec/api.yaml [hunk 3]
```diff
@@ -0,0 +1,269 @@
...
+        '500':
+          description: An unexpected error occurred
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/ErrorResponse"
+  /table/{table_name}/data:
+    get:
+      summary: Fetch data from a Table
+      description: Optional operation that returns data from a Table
```
Comment (jfuerth) [inline spec/api.yaml]: Is this one optional?

File: spec/api.yaml [hunk 4]
```diff
@@ -0,0 +1,269 @@
...
+        - Search API
+      responses:
+        '200':
+          description: Query results returned as TableData
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/TableData"
+        '400':
+          description: The query is invalid
```
Comment (jfuerth) [inline spec/api.yaml]: Should we change this to "Error in request headers or body, for example if the query is invalid. Details are provided in the ErrorResponse body."?

File: spec/api.yaml [hunk 5]
```diff
@@ -0,0 +1,269 @@
...
+      summary: 'Show information about the Data Connect service'
+      operationId: getServiceInfo
+      tags:
+        - Service Info API
+      responses:
+        '200':
+          description: |
+            Show information about this Data Connect service.
+
+            Use `"type": {"group": "org.ga4gh", "artifact": "data-connect", "version": "0.1.0"}` when implementing this specification directly.
```
Comment (jfuerth) [inline spec/api.yaml]: Version should be "1.0.0"

File: spec/api.yaml [hunk 6]
```diff
@@ -0,0 +1,269 @@
...
+          description: Optional description of the Table
+        data_model:
+          $ref: "http://json-schema.org/draft-07/schema#"
+        errors:
+          $ref: "#/components/schemas/ErrorList"
+      description: |
+        Describes a Table hosted by this Data Connect node.
+    TableData:
+      required:
+        - data_model
```
Comment (jfuerth) [inline spec/api.yaml]: We should remove this. The `data_model` is only required here if the `data` array is present. This was a late change to allow servers to implement "status polling via pagination". Should also add description text about this.

File: spec/api.yaml [hunk 7]
```diff
@@ -0,0 +1,269 @@
...
+            A short, human-readable description of the error.
+            The value should not change from occurrence to occurrence of an error, except for purposes of localization.
+          example: 'Internal server error'
+        detail:
+          type: string
+          description: 'A human-readable explanation specific to this occurrence of the error.'
+          example: 'Internal server error'
+      required:
+        - title
+    Pagination:
```
Comment (jfuerth) [inline spec/api.yaml:253]: Add link to pagination rules in SPEC.md

**Review by jfuerth — COMMENTED** [review]
Just a few loose ends that are probably worth cleaning up. I will send a new PR today to address my own comments here.

File: spec/api.yaml (hunk 3 above)
Comment (mcupak) [reply spec/api.yaml]: Huh no it shouldn't be.

File: spec/api.yaml (hunk 4 above)
Comment (mcupak) [reply spec/api.yaml]: Makes sense!

File: hugo/content/docs/reference/pagination-long-queries.md [hunk 8]
```diff
@@ -0,0 +1,129 @@
...
+POST /search
+content-type: application/json
+
+{"query":"select distinct gene_symbol from example_project.brca_exchange.v32"}
+
+HTTP/1.1 200 OK
+content-type: application/json
+retry-after: 1000
+
+{"data":[],"pagination":{"next_page_url":"/search/v1/statement/abc123/queued/1"}}
```
Comment (kemp-verily) [inline hugo/content/docs/reference/pagination-long-queries.md:58]: optional, but i think changing the URL here distracts a bit - there is no particular reason that the URL needs to change (this is a server implementation detail)

if you do want to show it, i would at least drop the queued/executing part and just keep the page numbers...

wdyt?

File: hugo/content/docs/reference/sql-functions.md [hunk 9]
```diff
@@ -0,0 +1,190 @@
...
+title: "SQL Functions"
+weight: 1
+draft: false
+lastmod: 2020-12-3
+type: docs
+layout: single-col
+---
+Data Connect's SQL dialect has been selected for compatibility with current major open source database platforms including Trino, PostgreSQL, and MySQL, and BigQuery. There are occasional name or signature differences, but a Data Connect implementation atop any of the major database platforms should be able to pass through queries that use the functions listed below with only minor tweaks.
+
+The functions below are a subset of those available in Trino 341. In a conformant Data Connect implementation, these functions must behave according to the Trino documentation. To assist with implementations directly on other database platforms, the [Trino Functions Support Matrix](https://docs.google.com/document/d/1y51qNuoe2ELX9kCOyQbFB4jihiKt2N8Qcd6-zzadIvk) captures the differences between platforms in granular detail.
```
Comment (kemp-verily) [inline hugo/content/docs/reference/sql-functions.md:11]: Is this net new or just moved?

I apologize if I missed the discussion but I am hesitant to make the claim that a conformant implementation needs to support everything here...

File: hugo/content/docs/security/data-source.md [hunk 10]
```diff
@@ -0,0 +1,77 @@
...
+lastmod: 2020-11-5
+# search related keywords
+type: docs
+layout: single-col
+---
+#### Securing Data Sources
+
+This page discusses different approaches to securing a Data Connect implementation depending on which implementation path you choose and how complex your access needs are.
+
+Data Connect can be implemented in many ways:
```
Comment (kemp-verily) [inline hugo/content/docs/security/data-source.md]: `many ways.  For example:`

File: hugo/content/docs/use-exisitng-data/tables-in-a-bucket/doc.md [hunk 11]
```diff
@@ -0,0 +1,53 @@
+---
+title: "Tables-in-a-bucket Method"
+weight: 1
+draft: false
+lastmod: 2020-11-5
+# search related keywords
+type: docs
+layout: single-col
+---
+#### What is a Phenopacket?
```
Comment (kemp-verily) [inline hugo/content/docs/use-exisitng-data/tables-in-a-bucket/doc.md]: Feels odd to open with this?  Can we move it down?

**Review by kemp-verily — COMMENTED** [review]
some nits; still need to review a couple things; sorry for the delay

File: hugo/content/docs/reference/sql-functions.md (hunk 9 above)
Comment (mcupak) [reply hugo/content/docs/reference/sql-functions.md:11]: Not net new, should be just lifted from SPEC.md. I'll check.

**Comment by mcupak** [comment]
@kemp-verily I agree with your comments. Are you going to send a PR or should I?

File: SPEC.md [hunk 12]
```diff
@@ -0,0 +1,1592 @@
...
+
+Data Connect specification is implementation-agnostic and does not prescribe use of a relational database or a particular database technology. However, for convenience, its SQL dialect has been selected for compatibility with current major open source database platforms including Trino, PostgreSQL, MySQL, and well as BigQuery. There are occasional name or signature differences, but a Data Connect API implementation atop any of the major database platforms should be able to pass through queries that use the functions listed below with only minor tweaks.
+
+With Trino (formerly PrestoSQL) being a popular database choice in Data Connect implementations, we've chosen its grammar as the basis for the grammar supported by Data Connect. Functions below are a subset of those available in Trino 341, and must behave according to the Trino documentation in a conforming Data Connect implementation. To assist with implementations directly on other database platforms, the [Trino Functions Support Matrix](https://docs.google.com/document/d/1y51qNuoe2ELX9kCOyQbFB4jihiKt2N8Qcd6-zzadIvk) captures the differences between platforms in granular detail. 
+
+*   **Logical Operators**
```
Comment (mcupak) [inline SPEC.md:444]: This entire section seems to have broken formatting, we should fix this. Not sure what the significance of `*` at the end of some lines is, some identifiers are not escaped or backticked correctly, the final section is off and JSON functions has broken formatting.

Comment (mcupak) [reply SPEC.md:444]: Looks automated. @jfuerth Do you remember where we transferred this from?

File: hugo/content/docs/reference/sql-functions.md (hunk 9 above)
Comment (mcupak) [reply hugo/content/docs/reference/sql-functions.md:11]: I took a few minutes now to go over the list of functions, and I'm comfortable with the "optional" approach now.

**Comment by mcupak** [comment]
Merging now that all the related issues have been resolved.

**Pull request merged**